src/pyhf/workspace.py

```diff
@@ -161,3 +163,186 @@ def data(self, model, with_aux=True):
         if with_aux:
             observed_data += model.config.auxdata
         return observed_data
+
+    def _prune_and_rename(
+        self,
+        prune_modifiers=[],
+        prune_modifier_types=[],
+        prune_samples=[],
+        prune_channels=[],
+        prune_measurements=[],
+        rename_modifiers={},
+        rename_samples={},
+        rename_channels={},
+        rename_measurements={},
+    ):
+        """
+        Return a new, pruned, renamed workspace specification. This will not modify the original workspace.
+
+        The pruned, renamed workspace must also be a valid workspace.
```

Comment:
I think we need to somewhere define what we mean by prunig, either here, or in the docs

Reply:
done.